cpp/src/arrow/memory_pool.cc

```diff
@@ -40,6 +40,14 @@ namespace arrow {
 constexpr size_t kAlignment = 64;
 
 namespace {
+
+#ifdef ARROW_JEMALLOC
+static size_t FixAllocationSize(int64_t size) {
```

Comment:
Could be inline (the compiler may inline it anyway as static + anonymous namespace)